x-pack/plugins/maps/public/classes/sources/es_geo_grid_source/create_source_editor.js

```diff
@@ -8,15 +8,25 @@ import _ from 'lodash';
 import React, { Fragment, Component } from 'react';
 import PropTypes from 'prop-types';
 
+import { ES_GEO_FIELD_TYPES } from '../../../../common/constants';
 import { SingleFieldSelect } from '../../../components/single_field_select';
 import { getIndexPatternService, getIndexPatternSelectComponent } from '../../../kibana_services';
 import { NoIndexPatternCallout } from '../../../components/no_index_pattern_callout';
 import { i18n } from '@kbn/i18n';
 
 import { EuiFormRow, EuiSpacer } from '@elastic/eui';
-import { getAggregatableGeoFieldTypes, getFieldsWithGeoTileAgg } from '../../../index_pattern_util';
+import {
+  getFieldsWithGeoTileAgg,
+  getGeoFields,
+  getGeoTileAggNotSupportedReason,
+  supportsGeoTileAgg,
+} from '../../../index_pattern_util';
 import { RenderAsSelect } from './render_as_select';
 
+function doesNotSupportGeoTileAgg(field) {
```

Comment:
consider just inlining as lambda iso creating named function

Reply:
Don't want to inline a function in react. That will create a new function instance every render and cause props to be different because functions do not equal. See number 2 in https://itnext.io/6-tips-for-better-react-performance-4329d12c126b